### DISCUSSION TITLE
Files downloaded by yarn containing binary content when it should be text.

### DISCUSSION THREAD

**Discussion by furier** [yarnpkg/berry#2933] [Q&A]
![image](https://user-images.githubusercontent.com/776919/119646625-acf60500-be1f-11eb-9680-0db737531521.png)

The picture describes it all, here are my steps to produce the outcome.

1. `git clone ...`
2. `yarn install`
3. `yarn start`

As you can see in the picture when i do the `ng serve` command the angular build fails because a javascript file is containing binary data and not text as it should. This happens constantly but which file it crashes on is never the same, just random, telling me there is not something specifically wrong with that file but with multiple files leading me to believe there is nothing wrong with the npm packages but rather the fetching of the files or the package manager, am I right?

Any ideas on how I can solve this or further debug to narrow down exactly what I need to fix?

Yarn: 1.22.5
Node.js: 12.22.1
npm: 6.14.12

--------------------------------------------------------------------------------------------------------------------------------------

I am sorry for the misplaced discussion, as it belongs in Yarn v1 repo. Created a new [issue](https://github.com/yarnpkg/yarn/issues/8639) over there FYI

**Comment by merceyz** [comment]
You're in the wrong repo, your issue is for Yarn 1 over at https://github.com/yarnpkg/yarn.

The issue is most likely that your cache has been corrupted, you can try clearing it with `yarn cache clean` https://classic.yarnpkg.com/en/docs/cli/cache#toc-yarn-cache-clean

> git clone ...

Git clone what exactly?

**Reply by furier** [reply]
@merceyz thanks for answering, I went to https://yarnpkg.com/ and just followed the github icon, didn't know this wasn't the correct repository 🤷‍♂️

The `git clone ...` refers to the repository where we use **yarn** as the package manager for our Angular project.

I do these commands every time I get this type of crash / error, which the last days have been constantly...

1. `rm -rf node_modules`
2. `yarn cache clean`
3. `yarn install`
4. `yarn start` aka `ng serve`

I have reinstalled Yarn using **npm** and **chocolatey** and it has persisted in some way, shape or form.

The Angular project just compiled successfully now but running a different `package.json` script `yarn run gql` aka `graphql-codegen --config codegen.schema.yml` produces error which are related

![image](https://user-images.githubusercontent.com/776919/119666215-77f4ad00-be35-11eb-9d19-f7cb210301e6.png)

The file is incomplete and contains something that looks like cyrillic to me, I have also encountered Chinese characters. I am totally lost 🤷‍♂️😥

**Reply by merceyz** [reply]
> I went to https://yarnpkg.com/ and just followed the github icon, didn't know this wasn't the correct repository 🤷‍♂️

The banner at the top says `This documentation covers Yarn 2. For 1.x docs, see classic.yarnpkg.com` but it's fine.

> The git clone ... refers to the repository where we use yarn as the package manager for our Angular project.

Clearly, the point was that we can't run `git clone ...` and somehow have git produce the same repository that you have.

> The file is incomplete and contains something that looks like cyrillic to me, I have also encountered Chinese characters. I am totally lost

That's interesting, if clearing the cache didn't help there isn't much we can do. You can try using the latest Yarn build and see if that helps but it looks like something just gets corrupted which it probably wouldn't fix.

```
yarn set version berry && yarn set version 3.0.0-rc.2
```

**Reply by furier** [reply]
Unfortunately I cannot provide the link to the repository, even with the link you would not be able to retrieve the code as it is proprietary code owned by by company in a private repository on azure. Thus all the attempts at censoring the company / project name etc.

The interesting bit to me is that clearing out everything and installing with **NPM** works fine 🤷‍♂️

**Comment by rally25rs** [comment]
very weird. I would try to track down the issue by doing:

```
yarn cache clean
yarn install --verbose > yarn.log
```
then check in the yarn log for where it downloaded `@babel/types` from. you should see a log line that looks like
```
verbose 3.89712084 Performing "GET" request to "https://registry.yarnpkg.com/@babel/types/-/types-7.14.2.tgz".
```
you could manually download that file with a `curl` request or just through a web browser and extract it to see if it looks right.

Also check the file after it was extracted to the yarn cache (your exact path may vary based on the babel-types version):
```
head `yarn cache dir`/npm-\@babel-types-7.14.2-4208ae003107ef8a057ea8333e56eb64d2f6a2c3-integrity/node_modules/\@babel/types/lib/modifications/typescript/removeTypeDuplicates.js
```

and then make sure that the file in the cache matches the file in your `node_modules` in the project itself.

**Reply by furier** [reply]
> then check in the yarn log for where it downloaded `@babel/types` from. you should see a log line that looks like
> 
> ```
> verbose 3.89712084 Performing "GET" request to "https://registry.yarnpkg.com/@babel/types/-/types-7.14.2.tgz".
> ```
> 
> you could manually download that file with a `curl` request or just through a web browser and extract it to see if it looks right.

I downloaded the file and extracted it with winrar and the content looks good.

> Also check the file after it was extracted to the yarn cache (your exact path may vary based on the babel-types version):
> 
> ```
> head `yarn cache dir`/npm-\@babel-types-7.14.2-4208ae003107ef8a057ea8333e56eb64d2f6a2c3-integrity/node_modules/\@babel/types
> ```

I also checked the yarn cache dir at `%LOCALAPPDATA%/Yarn/Cache/v6/` found the package dir and examined the content and it also looked fine.

The weirdest thing is that after doing the `npm install` I haven't been able to reproduce the issue with **yarn**. 🤦‍♂️
